mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraAnnotationMapper.java

```diff
@@ -0,0 +1,159 @@
+        });
+    }
+
+    public CassandraAnnotationMapper(Mailbox<CassandraId> mailbox, Session session, int maxRetry,
+            CodeInjector codeInjector) {
+        Preconditions.checkArgument(maxRetry > 0);
+        this.mailbox = mailbox;
+        this.session = session;
+        this.maxRetry = maxRetry;
+        this.codeInjector = codeInjector;
+    }
+
+    public MailboxAnnotation getAnnotation() {
+        ResultSet resultSet = getStoredAnnotationRow();
+        if (resultSet.isExhausted()) {
+            return SimpleMailboxAnnotation.EMPTY;
+        }
+        Map<MailboxAnnotationEntryKey, MailboxAnnotationEntryValue> annos = CassandraUtils.convertToStream(resultSet)
+                .map(row -> new AbstractMap.SimpleEntry<>(buildKey(row.getString(CassandraAnnotationTable.KEY)),
+                        buildValue(row.getString(CassandraAnnotationTable.VALUE))))
+                .collect(Collectors.toMap(AbstractMap.SimpleEntry::getKey, AbstractMap.SimpleEntry::getValue));
+
+        return new SimpleMailboxAnnotation(annos);
+    }
+
+    public void updateAnnotation(MailboxAnnotationCommand command) throws MailboxException {
+        try {
+            new FunctionRunnerWithRetry(maxRetry).execute(() -> {codeInjector.inject();
+                if (SimpleMailboxAnnotationEntryValue.NIL.equals(command.getEntryValue())) {
+                    deleteStoredAnnotation(command);
+                    return true;
+                } else {
+                    ResultSet resultSet = getStoredAnnotationRowByKey(command.getEntryKey())
+                            .map((row) -> updateStoredAnnotation(command, row))
+                            .orElseGet(() -> insertAnnotation(command));
+                    return resultSet.one().getBool(CassandraConstants.LIGHTWEIGHT_TRANSACTION_APPLIED);
+                }
+            });
+        } catch (LightweightTransactionException e) {
+            LOG.error("Exception during lightweight transaction: " + e.getMessage());
+            throw new MailboxException("Exception during lightweight transaction", e);
+        }
+    }
+
+    private ResultSet deleteStoredAnnotation(MailboxAnnotationCommand command) {
+        return session.execute(delete().from(CassandraAnnotationTable.TABLE_NAME)
+                .where(eq(CassandraAnnotationTable.ID, mailbox.getMailboxId().asUuid()))
+                .and(eq(CassandraAnnotationTable.KEY, command.getEntryKey().getName())).ifExists());
+    }
+
+    private ResultSet updateStoredAnnotation(MailboxAnnotationCommand command, Row row) {
+        return session.execute(update(CassandraAnnotationTable.TABLE_NAME)
+                .with(set(CassandraAnnotationTable.VALUE, command.getEntryValue().getValue()))
+                .where(eq(CassandraAnnotationTable.ID, mailbox.getMailboxId().asUuid()))
+                .and(eq(CassandraAnnotationTable.KEY, row.getString(CassandraAnnotationTable.KEY))).ifExists());
+    }
```

Comment:
Well, I would like this CRUD code also exposed on the MailboxMapper, and to the MailboxManager